lib/thrift/src/thrift/transport/TSSLServerSocket.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2006 Facebook
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cstring>
+#include <zephyr/net/tls_credentials.h>
+#include <zephyr/posix/sys/socket.h>
+#include <zephyr/posix/unistd.h>
+
+#include <thrift/thrift_export.h>
+#include <thrift/transport/TSSLServerSocket.h>
+#include <thrift/transport/TSSLSocket.h>
+
+#include <thrift/transport/TSocketUtils.h>
+
+template <class T>
+inline void* cast_sockopt(T* v) {
+  return reinterpret_cast<void*>(v);
+}
+
+void destroyer_of_fine_sockets(THRIFT_SOCKET* ssock);
+
+namespace apache {
+namespace thrift {
+namespace transport {
+
+enum ThriftTLScertificateType {
```

Comment:
I see why you had this in the public header now - because it's shared with `TSSLSocket.cpp`. 

You can put it in a common header "ThriftTLScertificateType.h" or something like that, and share it between the two files, but it's probably best to keep it out of the public / exported headers.